src/network/taraxa_capability.cpp

```diff
@@ -241,6 +246,8 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
           break;
         }
 
+        peer->passed_initial_ = true;
```

Comment:
Thinking about some edge case when the other node would restart very quickly in non-standard way (e.g. I would kill it with signal), would be such case detected and peer object deleted ? I guess even if not the other node sends initial packet  and it would not pass checks... Anyway just brainstorming if there is some sort of a situation when this flag would be still set to true even if it should not be as we do not set it to false anywhere...

Reply:
onDisconnect will erase peer object, so no worries there. Also keep in mind that your node needs to pass same check, so if the other node does not receive this packet from you, it will disconnect too

Reply:
I don't think such an edge case is really detectable.  A "split personality" node if you will could always maliciously connect in a valid way and then start sending garbage.   Handling such true maliciousness remains a TODO within taraxa capability, primely we don't stop syncing from the peer with most progress if they refuse to send us anything or they send us useless stuff.   Nor do we blacklist nodes for garbage. ...